base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseLifecycleManager.java

```diff
@@ -387,42 +391,62 @@ public void onReceived(RPCMessage message) {
-                                public void run() {
-                                    final String urlHttps = onSystemRequest.getUrl().replaceFirst("http://", "https://");
-                                    byte[] file = FileUtls.downloadFile(urlHttps);
-                                    if (file != null) {
-                                        SystemRequest systemRequest = new SystemRequest();
-                                        systemRequest.setFileName(onSystemRequest.getUrl());
-                                        systemRequest.setBulkData(file);
-                                        systemRequest.setRequestType(RequestType.ICON_URL);
-                                        if (isConnected()) {
-                                            sendRPCMessagePrivate(systemRequest, true);
+                                };
+                                handleOffboardTransmissionThread.start();
+                                return;
+                            } else if (requestType == RequestType.ICON_URL) {
+                                //Download the icon file and send SystemRequest RPC
+                                Thread handleOffBoardTransmissionThread = new Thread() {
+                                    @Override
+                                    public void run() {
+                                        final String urlHttps = onSystemRequest.getUrl().replaceFirst("http://", "https://");
+                                        byte[] file = FileUtls.downloadFile(urlHttps);
+                                        if (file != null) {
+                                            SystemRequest systemRequest = new SystemRequest();
+                                            systemRequest.setFileName(onSystemRequest.getUrl());
+                                            systemRequest.setBulkData(file);
+                                            systemRequest.setRequestType(RequestType.ICON_URL);
+                                            if (isConnected()) {
+                                                sendRPCMessagePrivate(systemRequest, true);
+                                            }
+                                        } else {
+                                            DebugTool.logError(TAG, "File was null at: " + urlHttps);
                                         }
-                                    } else {
-                                        DebugTool.logError(TAG, "File was null at: " + urlHttps);
                                     }
-                                }
-                            };
-                            handleOffBoardTransmissionThread.start();
+                                };
+                                handleOffBoardTransmissionThread.start();
+                                return;
+                            }
                         }
+
+                        DebugTool.logWarning(TAG, "Unable to handle OnSystemRequest");
```

Comment:
This logs a warning when we get a system request with type `LOCK_SCREEN_ICON_URL`